internal/discover/mounts_test.go

```diff
@@ -188,3 +188,131 @@ func TestMounts(t *testing.T) {
 		})
 	}
 }
+
+func TestMountsDeterministicOrder(t *testing.T) {
+	logger, _ := testlog.NewNullLogger()
+	mountOptions := []string{
+		"ro",
+		"nosuid",
+		"nodev",
+		"rbind",
+		"rprivate",
+	}
+
+	// Create a mock locator that returns paths in a non-deterministic order
+	mockLocator := &lookup.LocatorMock{
+		LocateFunc: func(s string) ([]string, error) {
+			// Return paths in different orders for each call
+			if s == "lib1" {
+				return []string{
+					"/usr/lib/libGLX_nvidia.so",
+					"/usr/lib/libGLX_indirect.so",
+					"/usr/lib/libnvidia.so",
+				}, nil
+			}
+			if s == "lib2" {
+				return []string{
+					"/usr/lib/libnvidia.so",
+					"/usr/lib/libGLX_indirect.so",
+					"/usr/lib/libGLX_nvidia.so",
+				}, nil
+			}
+			return []string{s}, nil
+		},
+	}
+
+	// Create two discoverers with the same configuration
+	d1 := &mounts{
+		logger:   logger,
+		lookup:   mockLocator,
+		required: []string{"lib1", "lib2", "lib3"},
+	}
+
+	d2 := &mounts{
+		logger:   logger,
+		lookup:   mockLocator,
+		required: []string{"lib1", "lib2", "lib3"},
+	}
+
+	// Get mounts from both discoverers
+	mounts1, err1 := d1.Mounts()
+	require.NoError(t, err1)
+
+	mounts2, err2 := d2.Mounts()
+	require.NoError(t, err2)
+
+	// Verify that both discoverers return the same mounts
+	require.Equal(t, len(mounts1), len(mounts2), "Both discoverers should return the same number of mounts")
+
+	// Verify that the mounts are in the same order
+	for i := range mounts1 {
+		require.Equal(t, mounts1[i].Path, mounts2[i].Path, "Mounts should be in the same order")
+		require.Equal(t, mounts1[i].HostPath, mounts2[i].HostPath, "Host paths should match")
+		require.Equal(t, mounts1[i].Options, mounts2[i].Options, "Mount options should match")
+		require.Equal(t, mountOptions, mounts1[i].Options, "Mount should have correct options")
+	}
```

Comment:
Does `require.EqualValues` not work here?

Reply:
edited to use EqualValues